documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+   $ openssl x509 -req -CA ca-cert.crt -CAkey ca-key.key \
+        -in server.csr -out server-cert.crt \
+        -days 365 -CAcreateserial -passin pass:changeit \
+        -extfile server-csr.conf -extensions v3_req
+   
+   # Optionally, check the contents of the signed certificate.
+   $ openssl x509 -in server-cert.crt -text -noout
+   ```
+
+4. Prepare a new keystore containing the signed server certificate and the CA's certificate chain. 
+
+   ```bash
+   # Import the CA certificate into a new keystore file.
+   $ keytool -keystore <server-jks-keystore-file-path> -alias <ca-alias> -noprompt \
+                -import -file <ca-cert-file-path> -storepass <server-jks-keystore-password>
+   
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore <server-jks-keystore-file-path> -alias <server-alias> -noprompt \
+                -import -file <signed-server-cert-file-path> -storepass <server-jks-keystore-password>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -keystore server.keystore.jks -alias CARoot -noprompt \
+             -import -file ca-cert.crt -storepass changeit
+
+   $ keytool -keystore server.keystore.jks -alias server -noprompt \
+                -import -file server-cert.crt -storepass changeit
+   
+   # Optionally, list the keystore and inspect the output
+   $ keytool -list -v -storepass changeit -keystore server.keystore.jks
+   ```
+
+5. Export the server certificate's private key. 
+   
+   This is a two-step process.
+   * First, convert the server's keystore in `.jks` format into `.p12` format.
+
+     ```bash
+     $ keytool -importkeystore \
+               -srckeystore <server-jks-keystore-file-path> \
+               -destkeystore <server-pkcs12-keystore-file-path> \
+               -srcstoretype jks -deststoretype pkcs12 \
+               -srcstorepass <server-jks-keystore-password> -deststorepass <server-pkcs12-keystore>
+     ```
+     
+     Example:
+     ```bash
+     $ keytool -importkeystore \
+               -srckeystore server.keystore.jks \
+               -destkeystore server.keystore.p12 \
+               -srcstoretype jks -deststoretype pkcs12 \
+               -srcstorepass changeit -deststorepass changeit
+     ```
+     
+   * Now, export the private key of the server into a `PEM` encoded file. Note that the generated `PEM` file is not protected
+     by a password. The key itself is password-protected, as we are using the `-nodes` flag. So, be sure
+     to protect it using operating system's technical and procedural controls.
+
+     ```bash
+     $ openssl pkcs12 -in <server-pkcs12-keystore-file-path> -passin pass:<key-password> \
+                      -nodes -out <server-key-file-path>
+     ```
+     
+     Example:
+     ```bash
+     $ openssl pkcs12 -in server.keystore.p12 -out server-key.key -passin pass:changeit -nodes
+     ```
+
+Step 5 concludes this stage and the stage is now set for installing the certificates and other TLS artifacts in Pravega.
```

Comment:
To reduce redundancy with the word "stage" in this line and the line below, consider:
"Step 5 concludes this stage and the stage is now set" -> "Step 5 concludes this stage and we are now set"

Reply:
Fixed.